src/visitors/pages/Login.js

```diff
@@ -0,0 +1,272 @@
+  align-items: center;
+  justify-content: center;
+`;
+
+const StyledSection = styled.section`
+  margin: ${props => props.margin}rem 0;
+  display: flex;
+  justify-content: center;
+`;
+
+const capitaliseFirstName = name => name.split(' ')[0].replace(/\b\w/g, l => l.toUpperCase());
+const greetingWithOrgName = orgName => `Welcome${orgName ? ` to ${orgName}` : ''}!`;
+
+export default class Login extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      initialFetchDone: false,
+      hasScanned: false,
+      visitorId: null,
+      visitorName: '',
+      orgName: '',
+      qrCodeContent: '',
+      activities: [],
+      errors: {},
+    };
+  }
+
+  componentDidMount() {
+    CbAdmin.downgradePermissions()
+      .then(() => Promise.all([
+        Activities.get({ day: 'today' }),
+        CommunityBusiness.get({ fields: ['name'] }),
+      ]))
+      .then(([resA, resC]) => {
+        this.setState({
+          initialFetchDone: true,
+          activities: resA.data.result,
+          orgName: resC.data.result.name,
+        });
+      })
+      .catch(error => redirectOnError(this.props.history.push, error));
+  }
+
+  onCameraError = () => {
+    this.props.history.push('/visitor/qrerror?e=camera');
+  }
+
+  onScannerError = () => {
+    this.props.history.push('/visitor/qrerror?e=scanner');
+  }
+
+  onSignInSuccess = (res) => {
+    this.setState(res);
+  }
+
+  onSignInFailure = () => {
+    this.props.history.push('/visitor/qrerror?e=no_user');
+  }
+
+  searchVisitorByQrCode = (content) => {
+    Visitors.search({ qrCode: content })
+      .then((res) => {
+        if (!res.data.result) {
+          throw new SignInForm.NoUserError('No user found');
+        }
+
+        return {
+          visitorName: res.data.result.name,
+          visitorId: res.data.result.id,
+          qrCodeContent: content,
+          hasScanned: true,
+        };
+      })
+      .catch(err => redirectOnError(this.props.history.push, err, { default: '/visitor/qrerror' }));
+  }
+
+  searchVisitorByDetails = filter =>
+    Visitors.get(null, { filter, fields: ['id', 'name'] })
+      .then(
+        (res) => {
+          if (!res.data.result || res.data.result.length === 0) {
+            throw new SignInForm.NoUserError('No user found');
+          }
+
+          if (res.data.result.length !== 1) {
+            throw new SignInForm.MultipleUserError('Multiple users found');
+          }
+
+          return {
+            visitorName: res.data.result[0].name,
+            visitorId: res.data.result[0].id,
+            hasScanned: true,
+          };
+        },
+        err => redirectOnError(this.props.history.push, err, { default: '/visitor/qrerror' }),
+      )
+
+  submitVisitorName = ({ name }) =>
+    this.searchVisitorByDetails({ name })
+
+  submitVisitorBirthYear = ({ name, birthYear }) =>
+    this.searchVisitorByDetails({
+      name,
+      age: birthYear ? [1, 1].map(() => moment().year() - birthYear) : undefined,
```

Comment:
THink about whether this makes sense as part of BirthYear/Age module, or think about adding birthYear as filter option.